STYLE-GUIDE.md

```diff
@@ -0,0 +1,109 @@
+<h1 dir="rtl">شیوه‌ نگارش</h1>
+
+<p dir="rtl">
+برای حفظ یکپارچگی ترجمه در تمام صفحات و همچنین میان سایر زبان‌ها، لطفا قوانین زیر را مدنظر داشته‌باشید.
+</p>
+
+<h2 dir="rtl">آی‌دی عنوان‌ها</h2>
+
+<p dir="rtl">هر عنوانی یک آی‌دی منحصربه‌فرد دارد، مانند زیر:</p>
+
+```md
+## Try React {#try-react}
+```
+<p dir="rtl">از ترجمه آی‌دی‌ها <b>بپرهیزید</b>! از آن‌ها برای هدایت به بخش‌های مختلف یک صفحه استفاده می‌شود. به خصوص اگر کاربر از یک لینک خارجی به یک بخش مورد نظر از صفحه هدایت شده‌باشد، این قابلیت از کار خواهد افتاد. برای مثال:</p>
+
+```md
+See the [beginning section](/getting-started#try-react) for more information.
+```
+
+<p dir="rtl">✅ صحیح:</p>
+
+```md
+## ری‌اکت را امتحان کنید {#try-react}
+```
+
+<p dir="rtl">❌ غلط:</p>
+
+```md
+## ری‌اکت را امتحان کنید {#react-ra-emtehan-konid}
+```
+<p dir="rtl">ترجمه باعث از کار افتادن لینک بالا شده‌است.</p>
+
+
+<h2 dir="rtl">متن در کد</h2>
+<p dir="rtl">
+فقط کامنت‌های یک کد را ترجمه کنید. شاید در بعضی موارد نیاز به ترجمه string باشد، اما نسبت به مواردی که ممکن است در منطق کد تغییر ایجاد کند آگاه باشید!
+</p>
+
+<p dir="rtl">مثال:</p>
+
+```js
+// Example
+const element = <h1>Hello, world</h1>;
+ReactDOM.render(element, document.getElementById('root'));
+```
+
+<p dir="rtl">✅ صحیح:</p>
+
+```js
+// مثال
+const element = <h1>Hello, world</h1>;
+ReactDOM.render(element, document.getElementById('root'));
+```
+
+<p dir="rtl">✅ صحیح:</p>
+
+```js
+// مثال
+const element = <h1>سلام، دنیا</h1>;
+ReactDOM.render(element, document.getElementById('root'));
+```
+
+<p dir="rtl">❌ غلط:</p>
+
+```js
+// مثال
+const element = <h1>سلام، دنیا</h1>;
+// "root" به آی‌دی یک تگ اشاره دارد
+// نباید آن را ترجمه کرد!
+ReactDOM.render(element, document.getElementById('ریشه'));
+```
+
+<p dir="rtl">❌ کاملا غلط:</p>
+
+```js
+// مثال
+const eleman = <h1>سلام، دنیا</h1>;
+ReactDOM.render(eleman, sanad.yaftanElemeanBaId('ریشه'));
+
+```
+<h2 dir="rtl">لینک‌های خارجی</h2>
+<p dir="rtl">
+اگر یک لینک خارجی به یک مقاله از منابعی مانند <a href="https://developer.mozilla.org/fa/" taget="_blank">شبکه توسعه‌دهنگان موزیلا (MDN)</a> و <a href="https://en.wikipedia.org/wiki/Main_Page" taget="_blank">ویکی‌پدیا (Wikipedia)</a>  ارجاع داده‌  شده‌است، در صورتی که نسخه فارسی آن مقاله موجود و از کیفیت مناسبی برخوردار است، به نسخه فارسی ارجاع داده‌شود.
+</p>
+
+<p dir="rtl">مثال:</p>
+
+<p>
+A stateful component inherits <code>Component</code> <a href="https://en.wikipedia.org/wiki/Class_(computer_programming)" target="_blank">class</a>.
+</p>
+
+
+
+<p dir="rtl">✅ صحیح:</p>
+<p dir="rtl">
+یک کامپوننت دارای state از <a href="https://fa.wikipedia.org/wiki/%DA%A9%D9%84%D8%A7%D8%B3_(%D8%A8%D8%B1%D9%86%D8%A7%D9%85%D9%87%E2%80%8C%D9%86%D9%88%DB%8C%D8%B3%DB%8C)" target="_blank">کلاس</a> <code>Component</code> ارث می‌برد.
```

Comment:
هدف اصلی ترجمه معادل سازی مناسب و موجز برای عبارتی مثل state هست. بکار بردن همان عبارت باعث تنزل برگردان به فارسی خواهدشد

Reply:
@masoud-bahrami  Some technical words do not have any proper translation in Persian context. In these cases, I myself prefer using the original word to avoid any confusion. Please, take a look at the following link. The simplified Chinese team used the same idea. They wrote a list of words that do not require to be translated.
[https://github.com/reactjs/zh-hans.reactjs.org/issues/2](https://github.com/reactjs/zh-hans.reactjs.org/issues/2)

Reply:
@sorousht  بله من هم موافق هستم با شما. برخی کلمات معادل سازی مناسبی نخواهند شد پس بهتره عین عبارت بکار برود.
اما در مورد `State` به نظرم توی برگردان ها جاهایی که مثلا قرار هست در مورد آن صحبت شه بهتر برگردانی شبیه
`حالت`  یا `وضعیت`
بکار برده شه. 
در مورد ترجمه چینی نمیشه بطور قطع نظر داد باید دید که برگردان این عبارت به چینی به چه صورت خواهد شد. ولی توی دایره لغت اونها هم بسیار موارد برگردان شده

Reply:
@masoud-bahrami @Schabaani @zahrajoulaei We have to discuss this challenging topic! :)

Reply:
I would agree to use English words as few as possible in the translated text, but it is inevitable that we can not find a replacement for all of them, here we also have a problem with the word "component" for example. My suggestion would be using a valid dictionary which includes a significant number of web/computer worlds translated to the Persian language.

Reply:
@sorousht @Schabaani  @zahrajoulaei  the most challenging part of translation is finding appropriate translation for terms such as `Declarative` .  As I said before glossary is very important, we should spend more time. I agree that the [Glossary]()  will be completed gradually. Bu at least it should contains more item in it :)

Reply:
[vachak](http://vachak.ir/ ) can be helpful. It has been developed by a couple of my friends who have had very good and useful experiences (for example [ضروریات اسکرام](https://www.digikala.com/product/dkp-124335/%DA%A9%D8%AA%D8%A7%D8%A8-%D8%A7%D8%B5%D9%88%D9%84-%D9%88-%D8%B1%D9%88%D8%B4-%DA%A9%D8%A7%D8%B1%D8%A8%D8%B1%D8%AF%DB%8C-%D8%A7%D8%B3%DA%A9%D8%B1%D8%A7%D9%85-%D8%A7%D8%AB%D8%B1-%DA%A9%D9%86%DB%8C-%D8%A7%D8%B3-%D8%B1%D9%88%D8%A8%DB%8C%D9%86-%D8%AC%D9%84%D8%AF-%D8%A7%D9%88%D9%84)) in translating computer works.

Reply:
@zahrajoulaei For the `component `word, the phrase `مولفه` is common in the community.

Reply:
When I hear component I imagine a small UI. But مولفه doesn't show it.

Reply:
@Schabaani `component `in the **Building Block** of React. It is the **Separation of Concern** for a `React Application` .

`Component` is a term that has come up with many definitions
As [Martin Fowler](https://martinfowler.com/articles/microservices.html#ComponentizationViaServices) said:

> When talking about components we run into the difficult definition of what makes a component.

There are different perceptions and definitions of the `component`. 
But the most accepted definition as Martin said is:

>  A component is a unit of software that is independently replaceable and upgradeable.

In this standpoint, the `مولفه` can be a proper term.
what's your opinion?